crates/searchers/openapi.yaml

```diff
@@ -0,0 +1,584 @@
+          $ref: "#/components/schemas/TokenAmount"
+        takerTokenFeeAmount:
+          $ref: "#/components/schemas/TokenAmount"
+
+    LiquidityParameters:
+      oneOf:
+        - $ref: "#/components/schemas/ConstantProductPool"
+        - $ref: "#/components/schemas/WeightedProductPool"
+        - $ref: "#/components/schemas/StablePool"
+        - $ref: "#/components/schemas/ConcentratedLiquidityPool"
+        - $ref: "#/components/schemas/ForeignLimitOrder"
+
+    Liquidity:
+      description: |
+        On-chain liquidity that can be used in a solution. This liquidity is
+        provided to facilitate TODO
+      allOf:
+        - $ref: "#/components/schemas/LiquidityParameters"
+        - type: object
+          properties:
+            address:
+              $ref: "#/components/schemas/Address"
+            gasEstimate:
+              type: integer
+              description: |
+                A rough approximation in the number gas units required to
+                use this liquidity on-chain.
+
+    Auction:
+      description: |
+        The abstract auction instance in order to solve.
+      type: object
+      properties:
+        id:
+          type: integer
+          description: |
+            The identifier for the auction. Will be set to `null` for requests
+            that are not part of an auction (when quoting token prices for
+            example).
+        tokens:
+          type: object
+          additionalProperties:
+            $ref: "#/components/schemas/Token"
+          description: |
+            A map of token addresses to token information.
+        orders:
+          type: array
+          items:
+            $ref: "#/components/schemas/Order"
+          description: |
+            The solvable orders included in the auction.
+        liquidity:
+          type: array
+          items:
+            $ref: "#/components/schemas/Liquidity"
+          description: |
+            On-chain liquidity that can be used by the solution.
+        gasPrice:
+          $ref: "#/components/schemas/TokenAmount"
+          description: |
+            The current estimated gas price. This is the price that will be used
+            for solution scoring.
+
+    JitOrder:
+      description: |
+        A just-in-time liquidity order included in a settlement.
+      type: object
+      properties:
+        sellToken:
+          $ref: "#/components/schemas/Token"
+        buyToken:
+          $ref: "#/components/schemas/Token"
+        receiver:
+          $ref: "#/components/schemas/Address"
+        sellAmount:
+          $ref: "#/components/schemas/TokenAmount"
+        buyAmount:
+          $ref: "#/components/schemas/TokenAmount"
+        validTo:
+          description: |
+            Unix timestamp until the order is valid to.
+          type: integer
+        appData:
+          $ref: "#/components/schemas/AppData"
+        feeAmount:
+          $ref: "#/components/schemas/TokenAmount"
+        kind:
```

Comment:
This one I am not sure about. I kept this 1:1 with the settlement contract signed order, because this is really a "signed protocol order", and not an abstraction. Otherwise, the data would not match the signature (noting that field names are part of the hashing process with EIP-712 signing scheme we use).

Reply:
Oh, that was a mistake. the type should be `OrderSide`.

Reply:
Oh, it already was `OrderSide`.